File: backend/app/schemas/performer_selection.py
from typing import Literal

from pydantic import BaseModel, Field


class RecommendationFallback(BaseModel):
    required_role: str
    recommended_sources: list[str]
    geography: str


class RecommendedPerformerItem(BaseModel):
    id: str
    full_name: str
    position: str
    organization: str | None = None
    is_internal: bool
    score: int = Field(ge=0, le=100)
    reasons: list[str]
    warnings: list[str]
    active_tasks: int = Field(ge=0)
    contact_available: bool


class PerformerRecommendationResponse(BaseModel):
    status: Literal["strong_match", "partial_match", "no_match"]
    confidence: int = Field(ge=0, le=100)
    recommended_performer_id: str | None = None
    performers: list[RecommendedPerformerItem]
    fallback: RecommendationFallback


class AssignRequestPayload(BaseModel):
    performer_id: str = Field(..., min_length=5)
    send_tz: bool = False
    contact_method: str = Field(default="email", min_length=2, max_length=50)
    recommended_performer_id: str | None = Field(
        default=None,
        description="Top recommendation id from GET /performers — used for override analytics.",
    )


class PerformerAnalyticsPayload(BaseModel):
    event: Literal["view", "assign", "override", "missing_contact"]
    performer_id: str | None = None
    payload: dict | None = None

File: src/types/performerSelection.ts
export type RecommendationStatus = 'strong_match' | 'partial_match' | 'no_match';

export interface RecommendationFallback {
  required_role: string;
  recommended_sources: string[];
  geography: string;
}

export interface RecommendedPerformerDTO {
  id: string;
  full_name: string;
  position: string;
  organization: string | null;
  is_internal: boolean;
  score: number;
  reasons: string[];
  warnings: string[];
  active_tasks: number;
  contact_available: boolean;
}

export interface PerformerRecommendationResponse {
  status: RecommendationStatus;
  confidence: number;
  recommended_performer_id: string | null;
  performers: RecommendedPerformerDTO[];
  fallback: RecommendationFallback;
}

export interface AssignPerformerPayload {
  performer_id: string;
  send_tz: boolean;
  contact_method: string;
  recommended_performer_id?: string | null;
}
